apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.ts

```diff
@@ -101,41 +114,73 @@ router.post(
+        throw new error.HttpStatusError(400, 'infoCourseInstance.json does not exist');
       }
-      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.id)) {
+      if (!req.body.ciid) {
+        throw new error.HttpStatusError(400, `Invalid CIID (was falsy): ${req.body.ciid}`);
+      }
+      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.ciid)) {
         throw new error.HttpStatusError(
           400,
-          `Invalid CIID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.id}`,
+          `Invalid CIID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.ciid}`,
         );
       }
+
+      const paths = getPaths(undefined, res.locals);
+
+      const courseInstanceInfo = JSON.parse(await fs.readFile(infoCourseInstancePath, 'utf8'));
+      courseInstanceInfo.longName = req.body.long_name;
+      const formattedJson = await formatJsonWithPrettier(JSON.stringify(courseInstanceInfo));
+
       let ciid_new;
       try {
-        ciid_new = path.normalize(req.body.id);
+        ciid_new = path.normalize(req.body.ciid);
       } catch {
         throw new error.HttpStatusError(
           400,
-          `Invalid CIID (could not be normalized): ${req.body.id}`,
+          `Invalid CIID (could not be normalized): ${req.body.ciid}`,
         );
       }
-      if (res.locals.course_instance.short_name === ciid_new) {
-        res.redirect(req.originalUrl);
-      } else {
-        const editor = new CourseInstanceRenameEditor({
+      const editor = new MultiEditor(
+        {
           locals: res.locals as any,
-          ciid_new,
-        });
-
-        const serverJob = await editor.prepareServerJob();
-        try {
-          await editor.executeWithServerJob(serverJob);
-          res.redirect(req.originalUrl);
-        } catch {
-          res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
-        }
+          description: `Update course instance: ${res.locals.course_instance.short_name}`,
+        },
+        [
+          new FileModifyEditor({
+            locals: res.locals as any,
+            container: {
+              rootPath: paths.rootPath,
+              invalidRootPaths: paths.invalidRootPaths,
+            },
+            filePath: infoCourseInstancePath,
+            editContents: b64EncodeUnicode(formattedJson),
+            origHash: req.body.orig_hash,
+          }),
+          new CourseInstanceRenameEditor({
+            locals: res.locals as any,
+            ciid_new,
+          }),
+        ],
+      );
+
+      const serverJob = await editor.prepareServerJob();
+      try {
+        await editor.executeWithServerJob(serverJob);
+      } catch {
+        return res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
       }
+      flash('success', 'Course instance configuration updated successfully');
+      return res.redirect(req.originalUrl);
```

Comment:
Nit, `return` shouldn't be necessary here.

```suggestion
     res.redirect(req.originalUrl);
```